packages/vite/src/content-for.ts

```diff
@@ -14,7 +14,7 @@ export function contentFor(): Plugin {
       let config: any = readJSONSync(join(locateEmbroiderWorkingDir(process.cwd()), 'content-for.json'));
       let contentsForConfig = config[path];
       for (const [contentType, htmlContent] of Object.entries(contentsForConfig)) {
-        html = html.replace(new RegExp(`{{content-for ['"]${contentType}['"]}}`), `${htmlContent}`);
+        html = html.replace(new RegExp(`{{content-for ['"]${RegExp.escape(contentType)}['"]}}`), `${htmlContent}`);
```

Comment:
escape is only supported on node 24

<img width="1004" height="441" alt="Image" src="https://github.com/user-attachments/assets/27de2f79-eb91-4f94-a782-a5c22297e7e9" />

Reply:
think that's why I was suggesting two replaces instead of RegExp

Reply:
Updated to use two replaces, and also updated the corresponding test.